### DISCUSSION TITLE
Issues importing bacpac - DB too small

### DISCUSSION THREAD

**Discussion by tarney76** [d365collaborative/d365fo.tools#822] [General]
I am having occasional issues importing a bacpac file to a tier 1 where the imported database will be much smaller than expected. Typically our db is approximately 40GB in size but when having problems it ranges closer to 1GB with all of our data missing. Oftentimes taking a new backup from LCS will resolve the issue but not always. If anyone has any guidance it would be much appreciated. The actual bacpac files themselves are the correct size. 

My PS line to import is "Import-D365Bacpac -ImportModeTier1 -BacpacFile $downloadPath -NewDatabaseName $newDBName -ModelFile $modelFileUpdatedPath -ShowOriginalProgress". This is after using Local-FixBacPacModelFile function to resolve the auto drop issue between different SQL version. (As mentioned here: https://community.dynamics.com/forums/thread/details/?threadid=72cf3065-4838-ee11-bdf4-000d3a574bff)

**Comment by FH-Inway** [comment]
Thanks for reporting. d365fo.tools uses SqlPackage for the .bacpac import. While we have not seen this particular issue before, it looks like it might not be caused by d365fo.tools.

To verify this, you can run the `Import-D365Bacpac` command with the `-OutputCommandOnly` parameter. This will write the SqlPackage.exe command that would be used to the command line. You can then check the command and execute it directly. If the same issue occurs, it is most likely not an issue with d365fo.tools.

To further analyze this, we will likely need more information from the import execution. Any additional output from the command that you can share would be helpful. There are also parameters `-LogPath` and `-DiagnosticFile` that create files with additional debug information. And there are some general [troubleshooting tips](https://github.com/d365collaborative/d365fo.tools/wiki/Troubleshoot) that might help collecting additional information.

Does this reproduce with the same .bacpac file when running the import multiple times?

Have you checked if there is enough disk space remaining for the import?

**Reply by tarney76** [reply]
Thanks for the reply. I have reached out to Microsoft as it was doing the same thing when importing direct to SSMS. Will wait to see what they say.

**Comment by FH-Inway** [comment]
I got the same issue. For me, it was caused by https://admin.powerplatform.microsoft.com/knownissues/4097694
Check the output from the import if it has any error messages. In this case, the database itself is created, but the error then prevents the data from being imported.
